tests/test_on_data/test_gin_ophys.py

```diff
@@ -5,17 +5,16 @@
 from roiextractors import NwbImagingExtractor, NwbSegmentationExtractor
 from roiextractors.testing import check_imaging_equal, check_segmentations_equal
 
-from neuroconv import (
-    NWBConverter,
-    ScanImageImagingInterface,
-    TiffImagingInterface,
-    Hdf5ImagingInterface,
-    SbxImagingInterface,
-    CaimanSegmentationInterface,
-    CnmfeSegmentationInterface,
-    ExtractSegmentationInterface,
-    Suite2pSegmentationInterface,
-)
+from neuroconv import NWBConverter
+
+from neuroconv.datainterfaces.ophys.scanimage import ScanImageImagingInterface
+from neuroconv.datainterfaces.ophys.tiff import TiffImagingInterface
+from neuroconv.datainterfaces.ophys.hdf5 import Hdf5ImagingInterface
+from neuroconv.datainterfaces.ophys.sbx import SbxImagingInterface
+from neuroconv.datainterfaces.ophys.caiman import CaimanSegmentationInterface
+from neuroconv.datainterfaces.ophys.cnmfe import CnmfeSegmentationInterface
+from neuroconv.datainterfaces.ophys.extract import ExtractSegmentationInterface
+from neuroconv.datainterfaces.ophys.suite2p import Suite2pSegmentationInterface
```

Comment:
My main concern with this approach is the massive break to back-compatibility that will ensue, which we said we wouldn't do again after the big tool/util modularization of https://github.com/catalystneuro/nwb-conversion-tools/pull/411

It's also a LOT of extra typing for a user (even with tab-completion, which not every IDE supports and not every new user might be aware of) especially if they are loading multiple interfaces at once.

I have a couple other propositions that work more under-the-hood to offer the same benefits, will open those PRs soon. I think we should try to consider different approaches and discuss a weighing of the benefits and drawbacks of each.

Reply:
Also see https://peps.python.org/pep-0690/ for extension discussion of both high-level and low-level details related this kind of problem and future improvements along the lines we're thinking.

Reply:
See #74 and #75 for some ideas...

Yet another approach to keep this PR nearly as-is but satisfy the backward compatibility breakage would be to override the module `__getattr__` which imports use (submodules are merely attributes of the main module, which is itself just a class). This is the suggestion of the accepted [PEP 562](https://peps.python.org/pep-0562/)

```python
# inside global `__init__.py`
# ...
mapping_from_data_interface_name_to_location_in_module: Dict[str, str]  # would obviously have a better name than this

def __getattr__(name):
    if name in mapping_from_data_interface_name_to_location_in_module:
        return importlib.import_module("." + mapping_from_data_interface_name_to_location_in_module[name], __name__)

# E.g.,
# >>> from neuroconv import DeepLabCutInterface
# 'essentially' calls
# DeepLabCutInterface = getattr(neuroconv, ".behavior.deeplabcut.deeplabcutinterface.DeepLabCutInterface")
```

This way, all imports from `neuroconv` would behave identically to how they did before, but the entire library of interfaces would only be loaded on-demand for each requested interface. Also all external imports in those interfaces would get called right away (PR #74 and #75 have the downside that import errors would only be revealed at initialization).

Reply:
Thanks for putting all those three options together and their implementation in PRs. It was educational for me. If we are going with any of these three options I favor the last one as it confines all the import logic to one file (the global init) and keeps the workflow identical everywhere else.